ci/ansible/base_image_ci.yaml

```
@@ -0,0 +1,72 @@
+# Purpose:
+# - To prepare a new base image with all packages required to run pulp playbooks
+# 
+# Actions:
+# - (ON RHEL) Subscribes to RHEL repos
+# - Installs all pre-requisite packages
+# - (IF enable_fips): Configures FIPS
+# - (ON RHEL IF rhel_clean) Unregisters RHEL subscription-manager
+#       - NOTE: All Pulp QE playbooks will fail if subscription manager is not disabled.
+#
+# Assumptions:
+# - If using RHEL, the `rhel_clean` must be set to `true`. Playbooks like `pulp-install` require no exsiting subscriptions
+# - For FIPS installs, `enable_fips` will configure FIPS as long as `/boot` or `/` exists
+# 
+# Result:
+# - A VM or KVM ready to run Pulp Ansible Playbooks
+# - For more information on how to export the image to Openstack, please see the Pulp QE Google Drive documentation
+#
+# Run Time:
+# - Non-FIPS:
+# - FIPS: 18 - 30 mins
+#
+# Example Usage:
+#
+# - Openstack with Cloud User and Setting FIPS on RHEL:
+#       - ansible-playbook -i 10.0.149.73, -u cloud-user base_image_ci.yaml -e "rhn_password=${RHN_PASSWORD}" -e "rhn_pool=${RHN_POOL}" -e "rhn_username=${RHN_USERNAME}" \
+#       -e "enable_fips=True" -e "rhel_clean=True"
+#
+# - Local VirtManager nstall on RHEL:
+#       - ansible-playbook -i 192.168.122.6, -u root base_image_ci.yaml -e "rhn_password=${RHN_PASSWORD}" -e "rhn_pool=${RHN_POOL}" -e "rhn_username=${RHN_USERNAME}" \
+#       -e "rhel_clean=True" 
+---
+- hosts: all
+  vars:
+    enable_fips: false
+    rhel_clean: false
+    ci_password: $6$mysecretsalt$MIJffjeQyfrKKrGkprGrDL/g2mCJa53koLmYQuuLmY9y37pDvGKPXU1Ov3RbMi.tpQ9cWvxAzUVtBLe7KrZoU.
+    ci_user: jenkins
+  roles:
+    - role: subscription-manager
+      when: ansible_distribution == 'RedHat'
+    - role: pulp-fips
+      when:
+        - enable_fips
+        - ansible_distribution == 'RedHat'
+    - role: jenkins-user-add
+  tasks:
+    - name: Install required packages
+      package:
+        name:
+          - java
+          - git
+          - ansible
+          - attr
+          - libselinux-python
+        state: present
+      become: yes
+
+    - name: Update all packages to latest
+      package:
+        name: "*"
+        state: latest
+      become: yes
+
+    - block:
+      - name: Unsubscribe all
+        command: subscription-manager unsubscribe --all
+
+      - name: Clear Cache
+        command: subscription-manager clean
+
+      when: rhel_clean
```

Could we simplify and just run this when it is a RHEL box? One less variable.

```suggestion
      when: ansible_distribution == 'RedHat'
```